service/history/historyEngine.go

```diff
@@ -2695,7 +2695,7 @@ func (e *historyEngineImpl) NotifyNewReplicationTasks(
 	tasks []persistence.Task,
 ) {
 
-	if len(tasks) > 0 {
+	if len(tasks) > 0 && e.replicatorProcessor != nil {
```

Comment:
Should it be a bug is the processor is nil?

Reply:
My understanding is this processor always set when engine initializes

Reply:
only set when global domain is enabled. notify replication for non-global cluster is the bug, processor being nil is not a bug.